legion/engine/llri-vk/detail/device.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file device.cpp
+ * @copyright 2021-2021 Leon Brands. All rights served.
+ * @license: https://github.com/Legion-Engine/Legion-LLRI/blob/main/LICENSE
+ */
+
+#include <llri/llri.hpp>
+#include <llri-vk/utils.hpp>
+
+namespace LLRI_NAMESPACE
+{
+    namespace internal
+    {
+        std::map<queue_type, uint32_t> findQueueFamilies(VkPhysicalDevice physicalDevice);
+    }
```

Comment:
It's not very clear that this is meant to be a shared function between multiple translation units. a more conventional way like having an internal vk only header that's included in the cpp file might be a little easier to understand/read.